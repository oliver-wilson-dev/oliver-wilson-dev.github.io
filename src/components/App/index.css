/**************************************************************
PROJECT LEVEL STYLING 
**************************************************************/
@import "../../styles/darkTheme.css";
@import "../../styles/lightTheme.css";

/*******************************
General
*******************************/
body {
  margin: 0;
	padding: 0;
	background-color: var(--primaryBackground);
}

.app {
  background-attachment: fixed;
  background-color: var(--primaryBackground);
  background-image: var(--primaryBackgroundImage);
  color: var(--primaryTextColor);
  display: flex;
  flex-direction: column;
  font-family: 'futura-pt', sans-serif;
  font-size: 1.2em;
  max-width: 100vw;
  min-height: 100vh;
}

.appContent {
	max-width: 100%;
	margin: 0;
  background: var(--primaryBackground);
  display: flex;
  flex-direction: column;
  justify-content: space-between;
  flex-grow: 1;
}

.cover-image {
  background-size: cover;
  background-position: center;
  background-repeat: no-repeat;
}

h1 {
  font-family: 'futura-pt', sans-serif;
}

h2, h3 {
  font-family: 'futura-pt', sans-serif;
  font-weight: 900;
  letter-spacing: 1px;
}

p {
  line-height: 2em;
  font-size: 16px;
}

a {
  text-decoration: none;
  font-family: 'futura-pt', sans-serif;
  color: var(--unvisitedLink);
}

.sectionTitle {
  font-family: 'futura-pt', sans-serif;
  font-weight: 900;
  font-size: 3em;
  padding: .5em 0 .5em 0;
  margin: 0;
  text-align: center;
}


/*******************************
flex
*******************************/
.flex-row {
  display: flex;
  flex-direction: row;
}

.flex-row-reverse {
  display: flex;
  flex-direction: row-reverse;
}

.flexColumn {
  display: flex;
	flex-direction: column;
	margin: 10px 30px;
}

.flexCenter {
  align-items: center;
  justify-content: center;
}

@media (min-width: 0px) and (max-width: 767px) {
  .sectionTitle {
		font-size: 28px;
	}
}

@media (min-width: 1024px) {
  .appContent {
		box-shadow: 8px 8px 0px #0b0a0a;
		border: 3px solid #0b0a0a;
		margin: 20px auto;
  }
}