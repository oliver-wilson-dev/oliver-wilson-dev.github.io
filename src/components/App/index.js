import React, {Component } from "react";
import styles from "./index.css";

class App extends Component {
    render() {
        console.log(styles);
        
        return (
            <div>
                <h1 className={styles.title}>Hello world</h1>
            </div>
        )
    }
}

export default App;