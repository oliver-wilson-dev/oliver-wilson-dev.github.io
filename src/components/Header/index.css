.header {
    top: 0;
    left: 0;
    z-index: 6;
    width: 100%;
    display: flex;
    position: fixed;
    align-items: center;
    box-sizing: border-box;
    justify-content: center;
    background-color: var(--primaryBackgroundLight);
    height: var(--headerHeight);
    padding: var(--headerPadding);
    box-shadow:  0px 0px 8px var(--primaryBorderColourLight);
}

.wrapper {
    width: 100%;
    height: 100%;
    display: flex;
    max-width: 850px;
    align-items: center;
    justify-content: space-between;
}

.icon {
    width: 30px;
    height: 30px;
}

.burgerMenu {
    @extend .icon;
}

.crossIcon {
    @extend .icon;
    position: absolute;
    top: 0;
    right: 0;
    margin: 10px;
    margin: var(--headerPadding);
    fill: var(--primaryTextColor);
    opacity: 0;

    transition-property: opacity;
    transition-duration: .3s;
    transition-timing-function: ease;
}

.crossIconShow {
    opacity: 1;
}

.menuContainer {
    width: 0px;
    background-color: transparent;

    height: 100vh;
    z-index: 5;
    position: absolute;
    top: 0;
    right: 0;

    display: flex;
    justify-content: flex-end;
}

@keyframes open {
    0% {
        width: 0px;
        background-color: transparent;
    }

    50% {
        width: 100vw;
        background-color: transparent;
    }

    100% {
        width: 100vw;
        background-color: var(--overlayBackground);
    }
}

@keyframes close {
    0% {
        width: 100vw;
        background-color: var(--overlayBackground);
    }

    50% {
        width: 100vw;
        background-color: transparent;
    }

    100% {
        width: 0px;
        background-color: transparent;
    }
}

.menuOpen {
    animation-name: open;
    animation-duration: 0.5s;
    animation-fill-mode: forwards;
    animation-direction: normal;
}

.menuClose {
    animation-name: close;
    animation-duration: 0.5s;
    animation-fill-mode: forwards;
    animation-direction: normal
}

.menuHideFirstRender {
    display: none;
}

.menuContent {
    width: 80%;
    display: flex;
    height: 100%;
    background-color: var(--primaryBackground);
    box-shadow: 0px 0px 8px var(--primaryBorderColourLight);
}

.menuList {
    width: 100%;
    padding: 0px;
    margin: 20px 0px;
    list-style-type: none;
    display: flex;
    flex-direction: column;
    justify-content: center;
}

.menuItem {
    text-align: center;
    text-align: center;
    font-weight: 700;
    font-size: 30px;
    font-family: 'Source Sans Pro', sans-serif;
    padding: 40px;

    &:hover {
        background-color: var(--primaryBackgroundInverse);
        color: var(--primaryTextColorInverse);
    }
}

.link {
    color: var(--primaryTextColor);
}

@media (min-width: 786px) {
    .menuContent {
        max-width: 600px;
    }
}