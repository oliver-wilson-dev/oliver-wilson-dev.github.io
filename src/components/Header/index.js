import React from 'react';
import PropTypes from 'prop-types';
import styles from './index.css';
import ThemeToggle from '../../containers/ThemeToggle';
import BurgerMenu from './burgerMenu.svg';
import SideNav from '../../containers/SideNav';

const Header = ({ toggleShowSideNav }) => (
  <nav
    className={styles.header}
  >
    <div className={styles.wrapper}>
      <ThemeToggle />
      <BurgerMenu className={styles.burgerMenu} onClick={toggleShowSideNav} />
    </div>
    <SideNav />
  </nav>
);

Header.propTypes = {
  toggleShowSideNav: PropTypes.func.isRequired
};


export default Header;
