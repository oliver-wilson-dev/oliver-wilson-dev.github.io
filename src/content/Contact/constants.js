export const EMAIL = 'oliver.wilson@asos.com';
export const PHONE_NUMBER = '07580081310';
