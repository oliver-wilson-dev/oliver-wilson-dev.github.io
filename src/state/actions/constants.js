// export const SECTIONS_API_ENDPOINT = 'https://express.oliver-wilson-dev.now.sh/sections';
export const SECTIONS_API_ENDPOINT = 'http://localhost:3000/sections/';
export const LIGHT_THEME = 'LIGHT_THEME';
export const DARK_THEME = 'DARK_THEME';

// ACTION TYPES
export const TOGGLE_THEME = 'TOGGLE_THEME';
export const FETCH_CONTENT = 'FETCH_CONTENT';
export const FETCH_CONTENT_FAILURE = 'FETCH_CONTENT_FAILURE';
