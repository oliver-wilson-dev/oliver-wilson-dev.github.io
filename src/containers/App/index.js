import { connect } from 'react-redux';
import App from '../../components/App';
import { getUseLightTheme } from '../../state/selectors';

const mapStateToProps = state => ({
  useLightTheme: getUseLightTheme(state)
});

export default connect(mapStateToProps, undefined)(App);
