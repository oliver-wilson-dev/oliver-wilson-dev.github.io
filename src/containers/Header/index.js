import { connect } from 'react-redux';
import Header from '../../components/Header';
import { getShowSideNav } from '../../state/selectors';
import toggleShowSideNav from '../../state/actions/toggleShowSideNav';

const mapStateToProps = state => ({
  showSideNav: getShowSideNav(state)
});

const mapDispatchToProps = dispatch => ({
  toggleShowSideNav: () => dispatch(toggleShowSideNav())
});

export default connect(mapStateToProps, mapDispatchToProps)(Header);
