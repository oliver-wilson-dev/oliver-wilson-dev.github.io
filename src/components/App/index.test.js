import React from 'react';
import { shallow } from 'enzyme';
import App from './index';
import Header from '../../containers/Header';
import { DARK_THEME } from '../../state/actions/constants';

jest.mock('react-router-dom', () => ({
  __esModule: true,
  Route: () => null,
  Switch: () => null
}));

jest.mock('../Router', () => {
  const Router = () => null;

  return Router;
});

jest.mock('../NotFound', () => {
  const NotFound = () => null;

  return NotFound;
});

jest.mock('../../containers/Header', () => {
  const Header = () => null;

  return Header;
});

jest.mock('../../containers/Footer', () => {
  const Footer = () => null;

  return Footer;
});

jest.mock('../../containers/CookieDisclaimer', () => {
  const CookieDisclaimer = () => null;

  return CookieDisclaimer;
});

jest.mock('../Landing', () => {
  const Landing = () => null;

  return Landing;
});

jest.mock('../../containers/Sections', () => {
  const Sections = () => null;

  return Sections;
});

const defaultProps = { theme: 'test-theme' };

const render = overrideProps => shallow(<App {...defaultProps} {...overrideProps} />);

describe('<App/> component', () => {
  it('should exist', () => {
    expect(render().exists()).toBe(true);
  });

  it('should render correctly', () => {
    expect(render()).toMatchSnapshot();
  });

  it('should render a div with the data-theme attribute equal to the value of the theme prop', () => {
    const mockTheme = 'test-theme';
    expect(render({ theme: mockTheme }).find('div').first().prop('data-theme')).toBe(mockTheme);
  });

  it('should render a <Header/> component', () => {
    expect(render().find(Header).exists()).toBe(true);
  });

  describe('when the theme is dark', () => {
    it('should render correctly', () => {
      expect(render({ theme: DARK_THEME })).toMatchSnapshot();
    });
  });
});
