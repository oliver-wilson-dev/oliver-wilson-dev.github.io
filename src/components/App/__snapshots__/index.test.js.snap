// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App/> component should render correctly 1`] = `
<div
  className="test-class-app"
>
  <div
    className="test-app-content"
  >
    <Landing />
    <Sections />
    <DownloadButton />
    <SocialMedia />
    <Footer />
  </div>
</div>
`;
